lib/astutils.cpp

```diff
@@ -238,6 +238,22 @@ bool isSameExpression(bool cpp, bool macro, const Token *tok1, const Token *tok2
     return commutativeEquals;
 }
 
+bool equalTokValue(const Token * const tok1, const Token * const tok2)
+{
+    return !tok1->values().empty() && !tok2->values().empty() &&
```

Comment:
how about:

    tok1->values() == tok2->values() && tok1->values().size()==1U && tok1->values().front().isKnown();

If the value is known then there is only 1 value in the list.

Reply:
Ah I see, so there isn't multiple known values?

Reply:
Doesn't something like this create multiple known values:

```cpp
if(a == 1 || a== 2) { a; } // Known values of 1 and 2
```

?

Reply:
currently it is only a known value if cppcheck can determine it's always a specific value.

but I agree somehow that `a` could have 2 known values (1 and 2) in that code. That would be better for this checker.

Reply:
So when there is multiple values they are treated as possible values, so I think I will just simplify it.